sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -617,10 +639,16 @@ case class CumeDist() extends RowNumberLike with SizeBasedWindowFunction {
   group = "window_funcs")
 // scalastyle:on line.size.limit line.contains.tab
 case class NthValue(input: Expression, offsetExpr: Expression, ignoreNulls: Boolean)
-    extends AggregateWindowFunction with ImplicitCastInputTypes {
+    extends AggregateWindowFunction with OffsetWindowSpec with ImplicitCastInputTypes {
 
   def this(child: Expression, offset: Expression) = this(child, offset, false)
 
+  override lazy val default = Literal.create(null, input.dataType)
+
+  override val direction = Ascending
+
+  override val isRelative = false
+
   override def children: Seq[Expression] = input :: offsetExpr :: Nil
 
   override val frame: WindowFrame = UnspecifiedFrame
```

Comment:
this is not needed as it's the same in `WindowFunction`

Reply:
`NthValue` extends `AggregateWindowFunction` and later override frame as:
`override val frame: WindowFrame = SpecifiedWindowFrame(RowFrame, UnboundedPreceding, CurrentRow)`